src/logic/exceptions/CodedException.ts

```diff
@@ -0,0 +1,37 @@
+import * as Sentry from '@sentry/react'
+import registry from './registry'
+import { IS_PRODUCTION } from 'src/utils/constants'
+
+export class CodedException extends Error {
+  public code: number
+  public uiMessage: string | undefined
+
+  constructor(code: keyof typeof registry, customMessage?: string) {
```

Comment:
the name `customMessage` suggests that this message will be used instead of the default one, but it is being concatenated to the actual message. Perhaps `extraMessage` could be a better name.